k2/csrc/log.cuh

```diff
@@ -69,12 +72,13 @@ class Logger {
     }
     printf("%s:%s:%u ", filename, func_name, line_num);
 #if defined(__CUDA_ARCH__)
-    printf("block:[%u,%u,%u], thread: [%u,%u,%u] ", blockIdx.x, blockIdx.y,
-           blockIdx.z, threadIdx.x, threadIdx.y, threadIdx.z);
+    printf("block:[%u,%u,%u], thread: [%u,%u,%u] ",
+           blockIdx.x, blockIdx.y, blockIdx.z,
+           threadIdx.x, threadIdx.y, threadIdx.z);
 #endif
   }
 
-  __host__ __device__ ~Logger() {
+  K2_CUDA_HOSTDEV ~Logger() {
```

Comment:
let's just leave as __host__ __device__ -> easier to read.

Reply:
there is an additional effect with it, for undefined __CUDA_ARCH__ it's empty, that help compiler.